qa/integration-tests/src/test/java/io/camunda/it/client/SearchUserTaskTest.java

```diff
@@ -0,0 +1,481 @@
+            .filter(f -> f.userTaskFollowUpDate(dateFilter))
+            .send()
+            .join();
+
+    assertThat(result.items().size()).isEqualTo(3);
+
+    // Check if the completion date is without the range
+    calendar.add(Calendar.DAY_OF_YEAR, -3);
+    dayBefore = dateFormat.format(calendar.getTime());
+
+    // Reset to current time
+    calendar.setTime(now);
+
+    // Calculate the date one day after
+    calendar.add(Calendar.DAY_OF_YEAR, 1);
+    dayAfter = dateFormat.format(calendar.getTime());
+
+    final DateFilter dateFilterOutOfRange = new DateFilter().from(dayBefore).to(dayAfter);
+    final var resultOutOfRange =
+        camundaClient
+            .newUserTaskQuery()
+            .filter(f -> f.userTaskFollowUpDate(dateFilterOutOfRange))
+            .send()
+            .join();
+    assertThat(resultOutOfRange.items().size()).isEqualTo(0);
+  }
+
+  @Test
+  public void shouldThrowAnExceptionWhenSearchBeforeAndAfterAreApplied() {
+    final List<String> searchData = List.of("1");
+
+    final CompletionException exception =
+        assertThrows(
+            CompletionException.class,
+            () -> {
+              camundaClient
+                  .newUserTaskQuery()
+                  .page(
+                      p ->
+                          p.searchAfter(Collections.singletonList(searchData))
+                              .searchBefore(Collections.singletonList(searchData)))
+                  .send()
+                  .join();
+            });
+
+    assertTrue(
+        exception
+            .getCause()
+            .getMessage()
+            .contains("Both searchAfter and searchBefore cannot be set at the same time"));
+  }
+
+  @Test
+  public void shouldValidatePagination() {
+    final var result = camundaClient.newUserTaskQuery().page(p -> p.limit(1)).send().join();
+    assertThat(result.items().size()).isEqualTo(1);
+    final var key = result.items().getFirst().getUserTaskKey();
+    // apply searchAfter
+    final var resultAfter =
+        camundaClient
+            .newUserTaskQuery()
+            .page(p -> p.searchAfter(Collections.singletonList(key)))
+            .send()
+            .join();
+
+    assertThat(resultAfter.items().size()).isEqualTo(2);
+    final var keyAfter = resultAfter.items().getFirst().getUserTaskKey();
+    // apply searchBefore
+    final var resultBefore =
+        camundaClient
+            .newUserTaskQuery()
+            .page(p -> p.searchBefore(Collections.singletonList(keyAfter)))
+            .send()
+            .join();
+    assertThat(result.items().size()).isEqualTo(1);
+    assertThat(resultBefore.items().getFirst().getUserTaskKey()).isEqualTo(key);
+  }
+
+  @Test
+  public void shouldSortTasksByStartDateASC() {
+    final var result =
+        camundaClient.newUserTaskQuery().sort(s -> s.creationDate().asc()).send().join();
+
+    assertThat(result.items().size()).isEqualTo(3);
+
+    // Retrieve IDs from the result
```

Comment:
💭 How about we just assert that creation date of item 0 is before that of item 1 and that the date of item 1 is before that of item 2? We don't have to extract and sort the whole list then. Just an idea.